src/env.c

```diff
@@ -1,6 +1,26 @@
 #include "rlang.h"
 
+
+bool r_is_unbound_value(SEXP x) {
+  return x == R_UnboundValue;
+}
+
 SEXP r_mut_env_parent(SEXP env, SEXP new_parent) {
```

Comment:
Maybe `#define mut`?

```c
r_mut_env_parent(mut sexp* env, sexp* new_parent);
```

That doesn't look bad but I'm not sure I like it because it seems to imply that `sexp` are otherwise const.